lib/rubygems/executor.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require 'open-uri'
+require 'json'
+require 'rubygems/platform'
+
+module Gem
+  module Web
+    class Executor
+
+      def supported_options_from_metadata
+        ["source_code_uri", "documentation_uri"]
+      end
+
+      def open_page(gem, options)
+        spec = Gem::Specification.find_by_name(gem)
+
+        if options[:sourcecode]
+          get_info_from_metadata(spec, "source_code_uri")
+        elsif options[:doc]
+          get_info_from_metadata(spec, "documentation_uri")
+        elsif options[:webpage]
+          open_default_browser(spec.homepage)
+        elsif options[:rubygems]
+          open_rubygems(gem)
+        else # The default option is homepage
+          open_default_browser(spec.homepage)
+        end
+      rescue Gem::MissingSpecError => e
+        puts e.message
+      end
+
+      def get_info_from_metadata(spec, info)
+        return unless supported_options_from_metadata.include?(info)
+
+        uri = spec.metadata[info]
+        if !uri.nil? && !uri.empty?
```

Comment:
1. It may be worth swapping this if/else statement around, to be consistent with `open_default_browser` below.
2. No need for parenthesis in the `puts(...)` call.

```ruby
if uri.nil? || uri.empty?
  puts "This gem does not have this information."
else
  open_default_browser(uri)
end
```

Reply:
Yeah, I thought about make it like this, but I was unsure. I'll update it!